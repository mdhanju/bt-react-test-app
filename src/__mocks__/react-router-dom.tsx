/* eslint-disable */

import React, { ReactNode } from 'react';

type Props = {
    children: ReactNode;
};

const rrd = require('react-router-dom');
// Just render plain div with its children
rrd.BrowserRouter = ({ children }: Props): ReactNode => <div>{children}</div>;
module.exports = rrd;
