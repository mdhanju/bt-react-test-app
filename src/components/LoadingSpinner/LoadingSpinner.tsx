import React from 'react'

import ContentHolder from '../ContentHolder'

const LoadingSpinner = (): React.ReactElement => {
    return (
        <ContentHolder>
            <div className="spinner-border" />
        </ContentHolder>
    )
}

export default LoadingSpinner
