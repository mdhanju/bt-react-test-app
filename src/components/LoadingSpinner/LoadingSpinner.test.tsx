import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import LoadingSpinner from './LoadingSpinner'

describe('<LoadingSpinner />', () => {
    it('should render properly without crashing', () => {
        const wrapper = shallow(<LoadingSpinner />)
        expect(wrapper.exists()).toBe(true)
        expect(wrapper.getElements()).toMatchSnapshot()
    })

    it('render the localized (En) Header', () => {
        const wrapper = render(
            <MemoryRouter>
                <LoadingSpinner />
            </MemoryRouter>
        )
        expect(wrapper.html()).toMatch('<div class="spinner-border"></div>')
    })
})
