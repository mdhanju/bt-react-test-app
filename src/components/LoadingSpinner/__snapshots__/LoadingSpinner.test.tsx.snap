// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingSpinner /> should render properly without crashing 1`] = `
Array [
  <ContentHolder>
    <div
      className="spinner-border"
    />
  </ContentHolder>,
]
`;
